packages/components/src/custom-select-control-v2/custom-select.tsx

```diff
@@ -113,13 +113,17 @@ function _CustomSelect(
 	return (
 		// Where should `restProps` be forwarded to?
 		<div className={ className }>
-			{ hideLabelFromVision ? ( // TODO: Replace with BaseControl
-				<VisuallyHidden as="label">{ label }</VisuallyHidden>
-			) : (
-				<Styled.SelectLabel store={ store }>
-					{ label }
-				</Styled.SelectLabel>
-			) }
+			<Styled.SelectLabel
+				store={ store }
+				render={
+					hideLabelFromVision ? (
+						// @ts-expect-error `children` are passed via the render prop
+						<VisuallyHidden as="label" />
+					) : undefined
+				}
+			>
+				{ label }
+			</Styled.SelectLabel>
```

Comment:
This change was cherry-picked from https://github.com/WordPress/gutenberg/pull/63137 just to speed up development on this PR.

I will wait for https://github.com/WordPress/gutenberg/pull/63137 to be merged, rebase and remove this change from this PR before merging.